### ISSUE TITLE
Is it possible to authenticate against Identity Provider (SAML or OIDC)?

### ISSUE THREAD

**Issue by MohammedNoureldin** [docker-mailserver/docker-mailserver#2983]
# Subject

I would like to configure a not documented mail server use case, particularly, to make Docker Mailserver authenticate against IdP using SAML or OIDC.

## Description

I am not sure how this can possible, that is why I am asking this question. Can we somehow configure Docker Mailserver to authenticate against SAML or OIDC?

The main benefit of this:
— We can only focus on configuring the IdP to use LDAP for example (or any other source) instead of making Docker Mailserver care about this. In the end.
— This allows the SSO feature.
— This will add MFA support out of the box, as it is supported by all IdP.
— So in other words, this will add support for many authentication use cases without exposing their complexity to Docker Mailserver.

I am not sure how the clients will prompt the webpage to finish the SSO authentication. So the whole workflow is still ambiguous for me, that is why I am opening this question, maybe we can figure it out together. IMHO, this will bring a huge benefit to Docker Mailserver project.

**Comment by polarathene** [comment]
This is on page 2 of open issues, please use search before raising a new issue :+1: 

https://github.com/docker-mailserver/docker-mailserver/issues/2713

I believe the issue author there is waiting on v12 release before they make an attempt (if they still have time for it).
